src/cbc_sdk/platform/policy_ruleconfigs.py

```diff
@@ -272,3 +299,316 @@ def set_assignment_mode(self, mode):
+        result = resp.json()
+        success = [d for d in result.get("successful", []) if d.get("id", None) == self.id]
+        if not success:
+            raise ApiError("update of host-based firewall failed")
+        self._info = success[0]
+        self._rule_groups = []
+        self._rule_groups_valid = False
+        self._mark_changed(False)
+
+    def _delete_ruleconfig(self):
+        """Perform the internal delete of the rule configuration object."""
+        my_id = self.id
+        self._cb.delete_object(self._base_url() + f"/{my_id}")
+        self._info = {"id": my_id}
+        self._full_init = False  # forcing _refresh() next time we read an attribute
+        self._rule_groups = []
+        self._rule_groups_valid = False
+        self._mark_changed(False)
+
+    @property
+    def enabled(self):
+        """
+        Returns whether or not the host-based firewall is enabled.
+
+        Returns:
+            bool: True if the host-based firewall is enabled, False if not.
+        """
+        return self.get_parameter('enable_host_based_firewall')
+
+    def set_enabled(self, flag):
+        """
+        Sets whether or not the host-based firewall is enabled.
+
+        Args:
+            flag (bool): True if the host-based firewall should be enabled, False if not.
+        """
+        self.set_parameter('enable_host_based_firewall', flag)
+
+    @property
+    def default_action(self):
+        """
+        Returns the default action of this rule configuration.
+
+        Returns:
+            str: The default action of this rule configuration, either "ALLOW" or "BLOCK."
+        """
+        default_rule = self.get_parameter('default_rule')
+        return default_rule.get("action", "ALLOW")
+
+    def set_default_action(self, action):
+        """
+        Sets the default action of this rule configuration.
+
+        Args:
+            action (str): The new default action of this rule configuration. Valid values are "ALLOW" and "BLOCK."
+        """
+        if action not in ("ALLOW", "BLOCK"):
+            raise ApiError(f"invalid default action: {action}")
+        default_rule = self.get_parameter('default_rule')
+        default_rule['action'] = action
+        self.set_parameter("default_rule", default_rule)
+
+    @property
+    def rule_groups(self):
+        """
+        Returns the list of rule groups in this rule configuration.
+
+        Returns:
+            list[FirewallRuleGroup]: The list of rule groups.
+        """
+        if not self._rule_groups_valid:
+            rg_param = self.get_parameter("rule_groups")
+            if rg_param is not None:
+                self._rule_groups = [HostBasedFirewallRuleConfig.FirewallRuleGroup(self._cb, d) for d in rg_param]
+                for group in self._rule_groups:
+                    group._init_parent(self)
+            else:
+                self._rule_groups = []
+            self._rule_groups_valid = True
+        return self._rule_groups
```

Comment:
```suggestion
        if self._info.get("rule_groups"):
             # includes parent initialization inside rule group constructor
             self._rule_groups = [HostBasedFirewallRuleConfig.FirewallRuleGroup(self._cb, d, self) for d in rg_param]
        return self._rule_groups
```

The additional `_rule_groups_valid` bool seems unnecesssary when we can check with `self._info.get("rule_groups")` as this returns false if the property is not present or if the list is empty

Reply:
The purpose of `_rule_groups_valid` is to signal that the value of `_rule_groups` is, in fact, valid, because the "lazy load" process has been done.  This is a pattern that I've used elsewhere in the SDK. Perhaps it should be named something like `_rule_groups_loaded`?

Aside from that, the change to use a constructor parameter instead of `_init_parent` has already been noted and will be done.

Reply:
I'm changing that variable to `_rule_groups_loaded` and also simplifying this function quite a bit by making the `get_parameter` method take a default value parameter. This will be in place at the next check-in.